### ISSUE TITLE
VsixCompatibilityAnalyzer target/task leaving file lock on the vsix file

### ISSUE THREAD

**Issue by holc-tkomp** [microsoft/VSSDK-Analyzers#153]
#### Note
This issue is actually about [Microsoft.VSSDK.BuildTools](https://www.nuget.org/packages/Microsoft.VSSDK.BuildTools/) and [Microsoft.VSSDK.CompatibilityAnalyzer](https://www.nuget.org/packages/Microsoft.VSSDK.CompatibilityAnalyzer) 17.2.x packages, but I was not able to find appropriate place for reporting issues with those packages.

#### Bug description
MSBuild target "VsixCompatibilityAnalyzer" from Microsoft.VisualStudio.Sdk.CompatibilityAnalyzerTasks.dll (Microsoft.VSSDK.CompatibilityAnalyzer package) is keeping file lock on the vsix file till the end of the **whole build**.

After updating Microsoft.VSSDK.BuildTools package from 17.1.4057 to 17.2.2186, our "AfterBuild" step, which performs custom signing of the vsix file, started to fail as it was not able to overwrite the vsix file.

#### To Reproduce
In our product, we call PowerShell script in "AfterBuild" target (cleaned up version):
```xml
<Target Name="AfterBuild">
  <CallTarget Targets="PublishArtifacts" />
</Target>
<Target Name="PublishArtifacts">
  <Exec Command="pwsh.exe -NonInteractive -NoProfile -ExecutionPolicy RemoteSigned -File &quot;$(MSBuildProjectDirectory)\Publish-Artifact.ps1&quot; -SourceArtifactPath &quot;$(MSBuildProjectDirectory)\$(TargetVsixContainer)&quot;" />
</Target>
```

In "Publish-Artifact.ps1", `Copy-Item` with `-Destination` being the vsix container will fail due to:
```
Copy-Item: D:\xyz\foobar.ps1:123
Copy-Item -Path $SomeCacheFile -Destination $VsixContainerPath -Force
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The process cannot access the file
'D:\xyz\bin\Release\net472\foobar.vsix' because it is being used by another process.
```

Example "Publish-Artifact.ps1", which should show the issue:
```powershell
param([string] $SourceArtifactPath)

function AwaitWriteable {
    [CmdletBinding()]
    param(
        [Parameter(Mandatory = $true)]
        [ValidateNotNullOrEmpty()]
        [string] $Path
    )

    $canWrite = $false
    $retryCount = 0

    while (!$canWrite -and $retryCount -lt 5) {
        $retryCount++
        try {
            [IO.File]::OpenWrite($Path).Close()
            $canWrite = $true
        }
        catch {
            if ($retryCount -lt 5) {
                Write-Host "Path '$Path' is not writeable. Retrying in 5 seconds ($retryCount/5)..."
            }
            Start-Sleep -Seconds 5
        }
    }

    if (!$canWrite) {
        throw "Unable to gain write access to '$Path'"
    }

    Write-Host "Got write access to '$Path'"
}

$global:ErrorActionPreference = 'Stop'

AwaitWriteable -Path "$SourceArtifactPath"
```

I added above code just to see if the file gets unlocked after few seconds, but it does not.

#### Expected behavior
"VsixCompatibilityAnalyzer" target/task should not leave file lock on the vsix file after it's done.

#### Additional context
Not sure if it's relevant, but the task reported some compatibilty issues:
```
6>Target VsixCompatibilityAnalyzer:
6>  Task "Message"
6>    Executing CompatibilityAnalyzer..
6>  Done executing task "Message".
6>  Using "VsixCompatibilityAnalyzer" task from assembly "D:\xyz\packages\microsoft.vssdk.compatibilityanalyzer\17.2.2186\build\Microsoft.VisualStudio.Sdk.CompatibilityAnalyzerTasks.dll".
6>  Task "VsixCompatibilityAnalyzer"
6>    D:\xyz\packages\microsoft.vssdk.compatibilityanalyzer\17.2.2186\build\Microsoft.VSSDK.CompatibilityAnalyzer.targets(40,5): warning VSIXCompatibility1001: The extension is incompatible with the targeted version of Visual Studio. More info at https://aka.ms/ExtensionSdkErrors. 
6>    D:\xyz\packages\microsoft.vssdk.compatibilityanalyzer\17.2.2186\build\Microsoft.VSSDK.CompatibilityAnalyzer.targets(40,5): warning VSIXCompatibility1001: Following references are incompatible.
6>    D:\xyz\packages\microsoft.vssdk.compatibilityanalyzer\17.2.2186\build\Microsoft.VSSDK.CompatibilityAnalyzer.targets(40,5): warning VSIXCompatibility1001: Microsoft.VisualStudio.Data.Schema.Package.dll -> Microsoft.VisualStudio.OLE.Interop; Microsoft.VisualStudio.Shell.Interop; Microsoft.VisualStudio.TextManager.Interop; Microsoft.VisualStudio.TextManager.Interop.8.0; Microsoft.VisualStudio.Shell.Interop.8.0; envdte80; envdte; VSLangProj; envdte; Microsoft.VisualStudio.Shell.Interop.10.0; Microsoft.VisualStudio.Shell.Interop.9.0; Microsoft.VisualStudio.Designer.Interfaces
6>  Done executing task "VsixCompatibilityAnalyzer".
6>Done building target "VsixCompatibilityAnalyzer" in project "VsPackage.2022.csproj".
```

**Comment by holc-tkomp** [comment]
Crude workaround (so that the vsix can be overwritten during build) is to skip the "VsixCompatibilityAnalyzer" target from ever running, by redefining it in .csproj:
```xml
  <Target Name="VsixCompatibilityAnalyzer">
  </Target>
```

**Comment by sajalverma17** [comment]
While I don't have any issues with the vsix file being locked (as I don't do any post build operation on it), I'd also like to know more about the warnings thrown by the `VsixCompatibilityAnalyzer` as mentioned in this issue.

```
6>  Task "VsixCompatibilityAnalyzer"
6>    D:\xyz\packages\microsoft.vssdk.compatibilityanalyzer\17.2.2186\build\Microsoft.VSSDK.CompatibilityAnalyzer.targets(40,5): warning VSIXCompatibility1001: The extension is incompatible with the targeted version of Visual Studio. More info at https://aka.ms/ExtensionSdkErrors. 
6>    D:\xyz\packages\microsoft.vssdk.compatibilityanalyzer\17.2.2186\build\Microsoft.VSSDK.CompatibilityAnalyzer.targets(40,5): warning VSIXCompatibility1001: Following references are incompatible.
6>    D:\xyz\packages\microsoft.vssdk.compatibilityanalyzer\17.2.2186\build\Microsoft.VSSDK.CompatibilityAnalyzer.targets(40,5): warning VSIXCompatibility1001: Microsoft.VisualStudio.Data.Schema.Package.dll -> Microsoft.VisualStudio.OLE.Interop; Microsoft.VisualStudio.Shell.Interop; Microsoft.VisualStudio.TextManager.Interop; Microsoft.VisualStudio.TextManager.Interop.8.0; Microsoft.VisualStudio.Shell.Interop.8.0; envdte80; envdte; VSLangProj; envdte; Microsoft.VisualStudio.Shell.Interop.10.0; Microsoft.VisualStudio.Shell.Interop.9.0; Microsoft.VisualStudio.Designer.Interfaces
6>  Done executing task "VsixCompatibilityAnalyzer".
```
I'm unable to understand what they are hinting at, and my extension does work fine despite them.

**Comment by AArnott** [comment]
The `VsixCompatibilityAnalyzer` doesn't come from this repo. The issue is tracked in https://developercommunity.visualstudio.com/t/VsixSignTool-randomly-failes-in-pipeline/10096652 and was fixed in the 17.3+ versions of the [Microsoft.VSSDK.BuildTools](https://www.nuget.org/packages/microsoft.vssdk.buildtools) package.
